feature-libs/checkout/base/components/checkout-review/checkout-review-overview/checkout-review-overview.component.ts

```diff
@@ -4,19 +4,68 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+} from '@angular/core';
 import { ActiveCartFacade, Cart } from '@spartacus/cart/base/root';
-import { Observable } from 'rxjs';
+import { FeatureConfigService, TranslationService } from '@spartacus/core';
+import { Observable, take } from 'rxjs';
 
 @Component({
   selector: 'cx-checkout-review-overview',
   templateUrl: './checkout-review-overview.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CheckoutReviewOverviewComponent {
+export class CheckoutReviewOverviewComponent implements AfterViewInit {
+  protected document = inject(DOCUMENT, { optional: true });
+  protected translationService = inject(TranslationService, { optional: true });
+  private featureService = inject(FeatureConfigService, { optional: true });
+
   constructor(protected activeCartFacade: ActiveCartFacade) {}
 
+  ngAfterViewInit(): void {
+    this.wrapComponentsWithSectionEl();
+  }
+
   get cart$(): Observable<Cart> {
     return this.activeCartFacade.getActive();
   }
+
+  /**
+   * Wraps checkout review components with section element required
+   * for applying correct a11y practices.
+   */
+  protected wrapComponentsWithSectionEl() {
+    if (
+      this.document &&
+      this.translationService &&
+      this.featureService?.isEnabled('a11yWrapReviewOrderInSection')
+    ) {
+      this.translationService
+        .translate('checkoutReview.reviewOrder')
+        .pipe(take(1))
+        .subscribe((label) => {
+          // We need to delay for a tick to let components render before querying.
+          setTimeout(() => {
```

Comment:
Maybe `requestAnimationFrame` would be a bit more suitable here? I think it should work the same way in this case, but I'd prefer `requestAnimationFrame` over `setTimeout`, since I guess timeout has a bit different purpose, however that's just my 2 cents